include/types.h

```diff
@@ -988,6 +989,21 @@ typedef struct link_speed
 
 } link_speed_t;
 
+// handling gzip files
+
+typedef struct hc_fp
+{
+  union
```

Comment:
Please remove union, it's just easier to read.

Reply:
mmm, sure? I think union here it's the right choise. If you don't like fp->f.fp we can add a new macro to access the member's of union.

Reply:
Yes, please remove it. We don't care about a few extra bytes at this point.

Reply:
ok